source/types/ut_assert_result_base.tps

```diff
@@ -0,0 +1,13 @@
+create or replace type ut_assert_result_base as object
+(
+  result   integer(1),
+  message  varchar2(4000 char),
+  expected varchar2(4000 char),
+  actual   varchar2(4000 char),
+  end_time timestamp with time zone,
+  not instantiable member procedure add_element( self in out nocopy ut_assert_result_base, an_assertion ut_assert_result_base ),
+  not instantiable member function get_element( a_position integer ) return ut_assert_result_base,
+  not instantiable member function get_elements_count return integer
+)
```

Comment:
I can't agree with the methods for the base abstraction. ut_assert_result_base is an element of assertions list for the test, it can't have methods oriented to the LIST, it can only have methods for itself. You can't ask an object for i-th element of the list if it dosn't content list at all and shouldn't contain it by the propose of it's class.

I think this methods shouldn't be in the assertion element base class.